src/modules/iotjs_module_websocket.c

```diff
@@ -110,14 +110,25 @@ static char *iotjs_ws_write_data(char *buff, void *data, size_t size) {
 }
 
 
-static bool iotjs_check_handshake_key(char *server_key) {
+static bool iotjs_check_handshake_key(char *server_key, jerry_value_t jsref) {
+  void *native_p;
+  JNativeInfoType *out_native_info;
+  bool has_p =
+      jerry_get_object_native_pointer(jsref, &native_p, &out_native_info);
+  if (!has_p || out_native_info != &wsclient_native_info) {
+    return false;
+  }
+  iotjs_wsclient_t *wsclient = (iotjs_wsclient_t *)native_p;
+
   unsigned char *out_buff = NULL;
 
-  size_t concatenated_size = strlen(WS_GUID) + strlen((char *)generated_key);
+  size_t concatenated_size =
+      strlen(WS_GUID) + strlen((char *)wsclient->generated_key);
```

Comment:
We should use `strnlen` in case of the `generate_key` to maximize the key size and avoid buffer overruns.